### PR TITLE
feat: implement articles audio

### PR DESCRIPTION
**Pull request by shahin-hq**
## Summary
This PR handles first portion of [this task](https://app.clickup.com/t/8678v0juv): implements feature to get article audio generated. I left out the frontend side of things for a separate PR in order to make PR reviews easy.

Most of the code are copied (as requested in the task description) from the msq.io repo while at the same time I had touch specific parts to match changes in `dashbrd`.
I tested audio generation part with Amazon Polly, it should work properly.

## Checklist

-   [ ] I checked my UI changes against the design and there are no notable differences
-   [ ] I checked my UI changes for any responsiveness issues
-   [x] I checked my (code) changes for obvious issues, debug statements and commented code
-   [ ] I opened a corresponding card on Clickup for any remaining TODOs in my code
-   [ ] I added a short description on how to test this PR _(if necessary)_
-   [ ] I added a storybook entry for the component that was added _(if necessary)_
-   [ ] Documentation _(if necessary)_
-   [x] Tests _(if necessary)_
-   [x] Ready to be merged

### CODE REVIEW

File: app/Observers/ArticleObserver.php [hunk 1]
```diff
@@ -0,0 +1,42 @@
+<?php
```
Comment (shahin-hq) [inline app/Observers/ArticleObserver.php:1]: Decided to keep observer related stuff in a separate class opposed to the msq implementation for the sake of simplicity.

File: database/seeders/ArticleSeeder.php [hunk 2]
```diff
@@ -21,13 +22,14 @@ public function run(): void
...
         if (! $network) {
             $network = Network::factory()->create();
-
         }
 
         $authors[] = User::factory()->editor()->withAvatar()->create();
         $authors[] = User::factory()->editor()->withAvatar()->create();
         $authors[] = User::factory()->editor()->create();
 
+        ConvertArticleToSpeech::disable();
```
Comment (shahin-hq) [inline database/seeders/ArticleSeeder.php:31]: Disabled audio conversion in seeder.

File: app/Console/Commands/GenerateAudioFileForArticle.php [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+   ';
+
+    protected $description = 'Regenerate audio file for the article with the given ID';
+
+    public function handle(): int
+    {
+        if (! ConvertArticleToSpeech::$enabled || ! (bool) config('dashbrd.text_to_speech.enabled')) {
+            $this->warn('Article text-to-speech is currently disabled.');
+
+            return 0;
```
Comment (alfonsobries) [inline app/Console/Commands/GenerateAudioFileForArticle.php:28]:
```suggestion
            return Command::SUCCESS;
```

File: app/Console/Commands/GenerateAudioFileForArticle.php [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+
+        $articles = $this->articles()
+                    ->isPublished()
+                    ->when($this->option('missing'), static fn ($query) => $query->whereNull('audio_file_url'))
+                    ->get();
+
+        if ($articles->isEmpty()) {
+            $this->warn('No articles found.');
+
+            return 0;
```
Comment (alfonsobries) [inline app/Console/Commands/GenerateAudioFileForArticle.php:39]:
```suggestion
            return Command::SUCCESS;
```

File: app/Console/Commands/GenerateAudioFileForArticle.php [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+            ConvertArticleToSpeech::dispatch($article);
+        });
+
+        $this->info('Regenerating audio files for '.$articles->count().' articles.');
+
+        $chars = $articles->sum(static fn ($article) => strlen($article->content));
+
+        $this->info('Total number of characters sent to AWS: '.number_format($chars).' (could be less because this also includes markdown characters).');
+
+        return 0;
```
Comment (alfonsobries) [inline app/Console/Commands/GenerateAudioFileForArticle.php:52]:
```suggestion
        return Command::SUCCESS;
```

File: app/Observers/ArticleObserver.php [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+            return;
+        }
+
+        if (! ($article->wasRecentlyCreated || $article->wasChanged(['content', 'is_published']))) {
+            return;
+        }
+
+        if (! ConvertArticleToSpeech::$enabled) {
+            return;
+        }
```
Comment (alfonsobries) [inline app/Observers/ArticleObserver.php:38]: I would move this as the first condition, if not enabled the is no reason to check if the article changed and stuff like that

**Review by alfonsobries — APPROVED** [review]
some style suggestion but other than that seems ok

**Review by crnkovic — APPROVED** [review]

**Comment by ItsANameToo** [comment]
may need adjustments related to the ACL call

**Review by ItsANameToo — APPROVED** [review]

**Pull request merged**